src/VirtoCommerce.InventoryModule.Web/Controllers/Api/InventoryModuleController.cs

```diff
@@ -235,8 +224,76 @@ public async Task<ActionResult<InventoryInfo>> UpdateProductInventory([FromBody]
         [Authorize(Permissions.Update)]
         public async Task<ActionResult> UpsertProductInventories([FromBody] InventoryInfo[] inventories)
         {
-            await _inventoryService.SaveChangesAsync(inventories);
+            await inventoryService.SaveChangesAsync(inventories);
             return Ok();
         }
+
+        /// <summary>
+        /// Partial update for the specified FulfillmentCenter by id
+        /// </summary>
+        /// <param name="id">FulfillmentCenter id</param>
+        /// <param name="patchDocument">JsonPatchDocument object with fields to update</param>
+        [HttpPatch]
+        [Route("inventory/fulfillmentcenters/{id}")]
+        [Authorize(Permissions.FulfillmentEdit)]
+        [ProducesResponseType(typeof(void), StatusCodes.Status204NoContent)]
+        public async Task<ActionResult> PatchFulfillmentCenter(string id, [FromBody] JsonPatchDocument<FulfillmentCenter> patchDocument)
+        {
+            if (patchDocument == null)
+            {
+                return BadRequest();
+            }
+
+            var fulfillment = await fulfillmentCenterService.GetByIdAsync(id);
+            if (fulfillment == null)
+            {
+                return NotFound();
+            }
+
+            patchDocument.ApplyTo(fulfillment, ModelState);
+
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            await fulfillmentCenterService.SaveChangesAsync([fulfillment]);
+
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Partial update for the specified inventory of product by id
+        /// </summary>
+        /// <param name="id">Inventory of product id</param>
+        /// <param name="patchDocument">JsonPatchDocument object with fields to update</param>
+        [HttpPatch]
+        [Route("inventory/products/{id}")]
```

Comment:
The route `inventory/products/{id}` means that `id` is a product id, but it's being used as inventory id.